frontend/src/pages/news/insideNewsPage.jsx

```diff
@@ -34,14 +37,156 @@ export default function InsideNewsPage() {
+    }, [initialLikeStatus]);
+
+    useEffect(() => {
+        likeStatusRef.current = likeStatus;
+    }, [likeStatus]);
+
+    const handleLikeStatus = async () => {
+        if (likeStatusRef.current !== initialLikeStatusRef.current) {
+            let interactionType = likeStatusRef.current ? 'like' : 'dislike';
+            saveInteraction(interactionType);
+        }
+    };
+
+    useEffect(() => {
+        const logInteraction = async () => {
+            await saveInteraction('open');
+        };
+        logInteraction();
+    }, []);
+
+    const saveInteraction = async (interactionType) => {
+        if (user) {
+            try {
+                await axios.post(
+                    `${apiEndpoints.url}${apiEndpoints.interactions.saveInteraction}`,
+                    {
+                        user_id: user.id,
+                        blob_id: articleId,
+                        interaction_type: interactionType,
+                    }
+                );
+            } catch (error) {
+                toast.error(`Troubles with saving interaction: ${error}`)
+            }
+        }
+    };
+
+    useEffect(() => {
+        async function getInteractionsCounts() {
+            try {
+                const { data } = await axios.get(`${apiEndpoints.url}${apiEndpoints.interactions.getInteractionsCounts}`, {
+                    params: {
+                        blob_id: articleId,
+                    },
+                });
+                setLikes(data.likes);
+                setViews(data.views);
+            }
+            catch (error) {
+                toast.error(`Troubles with getting interactions count: ${error}`)
+            }
+        }
+        getInteractionsCounts()
+    }, [] );
+
+    useEffect(() => {
+        async function getLikeStatus() {
+            try {
+                const { data } = await axios.get(`${apiEndpoints.url}${apiEndpoints.interactions.getInteractionStatus}`, {
+                    params: {
+                        user_id: user.id,
+                        blob_id: articleId,
+                        interaction_type: 'like'
+                    },
+                });
+                setLikeStatus(data);
+                setInitialLikeStatus(data);
+            } catch (error) {
+                toast.error(`Troubles with getting like status: ${error}`)
+            }
+        }
+
+        if (user && articleId) {
+            getLikeStatus();
+        }
+    }, [user, articleId]);
+
+    const toggleLike = () => {
+        if (user) {
+            setLikeStatus(prev => {
+                const newStatus = !prev;
+                setLikes(count => newStatus ? count + 1 : count - 1);
+                return newStatus;
+            });
+        }
+        else {
+            const notify = () => toast('Sign in to leave your reaction', {
+                action: {
+                    label: 'sign in',
+                    onClick: () =>  navigate('/sign-in') ,
```

Comment:
I would also like to see this "/sign-in" specified somewhere in some sort of a variable or whatever.

Imagine our project grows and has dozens of places with login page readdress. And then imagine we need to rename it from sign-in to something else. It's much better to have one place to change than 10+, is it not?

Talk this through with The Boys to decide on the best place to put that stuff

Reply:
We discussed it on one of our meetings: those links (e.g. sign-up in our case) should be defined somewhere (in one place). Ask the guys. I believe they have already put it somewhere